stdlib/array.ncl

```diff
@@ -120,14 +139,14 @@
           in
           go acc 0,
 
-    fold : forall a b. (a -> b -> b) -> b -> Array a -> b
+    fold_right : forall a b. (a -> b -> b) -> b -> Array a -> b
       | doc m%"
         Fold a function over a array.
```

Comment:
I have actually no idea what "fold" means, and this sentence doesn't help me understanding what it's doing. Is it simply common knowledge I'm missing?

Reply:
If I remember right, we had a discussion during the Nickel hour that made these changes about that. We couldn't come up with a good, concise explanation on the spot and left it this way.

While I think most functional programmers will know what "folds" are, we probably should include some more explanation in this documentation. I'm just not sure what to write exactly 😅

Reply:
For the record, here is Ocaml's documentation for `fold_right`

> f`old_right f [a1; ...; an] init` is `f a1 (f a2 (... (f an init) ...))`. Not tail-recursive.

Reply:
Actually, we've used something similar from the beginning, because I also find the documentation of Ocaml's folds to be quite good :slightly_smiling_face: . If you scroll down this file, the full documentation in Nickel is in fact:

>Fold a function over a array.
>`fold f init [x1, x2, ..., xn]` results in `f x1 (f x2 (... (f xn init) ...))`.
>
>For example:
>```
>  fold (fun e acc => acc @ [e]) [] [ 1, 2, 3 ] =>
>    ((([] @ [3]) @ [2]) @ [1]) =>                
>    [ 3, 2, 1 ]
>```

Reply:
Revisiting this, I think we can still expand a bit for non functional programmers. Maybe add a line or two explaining that fold are more or less like `for` loops, with examples. @rapenne-s has a point that `fold` might not be something very intuitive to DevOps people.